engine-types/src/parameters/connector.rs

```diff
@@ -220,6 +220,36 @@ impl rlp::Encodable for LogEntry {
     }
 }
 
+/// Borsh-encoded parameters for `set_erc20_metadata` function.
+#[derive(BorshSerialize, BorshDeserialize, Debug, Eq, PartialEq, Clone)]
+pub struct SetErc20MetadataArgs {
+    /// Address of the ERC-20 contract.
+    pub erc20_address: Address,
+    /// Metadata of the ERC-20 contract.
+    pub erc20_metadata: Erc20Metadata,
+}
+
+/// Metadata of ERC-20 contract.
+#[derive(BorshSerialize, BorshDeserialize, Debug, Eq, PartialEq, Clone)]
+pub struct Erc20Metadata {
+    /// Name of the token.
+    pub name: String,
+    /// Symbol of the token.
+    pub symbol: String,
+    /// Number of decimals.
+    pub decimals: u8,
+}
+
+impl Default for Erc20Metadata {
+    fn default() -> Self {
+        Self {
+            name: "Token".to_string(),
+            symbol: "TKN".to_string(),
+            decimals: 18, // To be compatible with MetaMask.
+        }
```

Comment:
The original default values were `"Empty", "EMPTY", 0`. Should we leave them as they were? cc @karim-en

Reply:
At least these changes don't affect our tests at all 😄. Also, I thought about adding a possibility to set optional metadata while deploying and mirroring a new ERC-20 contract.

Reply:
I think we should keep the `"Empty", "EMPTY", 0` as a default metadata.